pkg/controller/sparkapplication/controller.go

```diff
@@ -680,9 +675,18 @@ func (c *Controller) submitSparkApplication(app *v1beta2.SparkApplication) *v1be
 		}
 	}
 
-	submissionID, driverPodName, err := c.subJobManager.createSubmissionJob(app)
+	var submissionID string
+	var driverPodName string
+	var err error
+
+	if app.Spec.Mode == v1beta2.ClientMode {
+		submissionID, driverPodName, err = c.clientSubManager.createClientDriverPod(app)
+	} else {
+		submissionID, driverPodName, err = c.subJobManager.createSubmissionJob(app)
+	}
+
 	if err != nil {
-		if !errors.IsAlreadyExists(err) {
+		if !errors.IsAlreadyExists(err) || app.Spec.Mode == v1beta2.ClientMode {
```

Comment:
Why do we need this extra condition?

Reply:
If it errors because the driver pod already exists it should not ignore that and still change to failed since we want retries of this pod on the flyte side.